File: FiveLab/app/src/main/java/com/example/fivelab/AddFragment.java
package com.example.fivelab;

import android.content.ContentValues;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.net.Uri;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.databinding.DataBindingUtil;
import android.provider.MediaStore;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.CalendarView;
import android.widget.CheckBox;
import android.widget.DatePicker;
import android.widget.Spinner;
import android.widget.Toast;
import com.example.fivelab.databinding.FragmentAddBinding;
import java.io.File;
import java.sql.Date;
import java.util.Calendar;

public class AddFragment extends Fragment {

    private static final String TAG = "ExceptionLog";

    public AddFragment() {
    }
    DataBaseHelper helper;
    SQLiteDatabase db;
    Task task=new Task();
    View rootView;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        FragmentAddBinding binding=DataBindingUtil.inflate(inflater,R.layout.fragment_add,
                container,false);
        rootView=binding.getRoot();
        binding.setTask(task);
        ArrayAdapter<CharSequence> adapter=ArrayAdapter.createFromResource(rootView.getContext(),R.array.ComplexityArray, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        Spinner ComplexitySpinner=rootView.findViewById(R.id.ComplexitySpinner);
        ComplexitySpinner.setAdapter(adapter);
        adapter=ArrayAdapter.createFromResource(rootView.getContext(),R.array.UrgencyArray, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        ComplexitySpinner=rootView.findViewById(R.id.UrgencySpinner);
        ComplexitySpinner.setAdapter(adapter);
        CalendarView datePicker=rootView.findViewById(R.id.Deadline);
        datePicker.setOnDateChangeListener(new CalendarView.OnDateChangeListener() {
            @Override
            public void onSelectedDayChange(@NonNull CalendarView calendarView, int i, int i1, int i2) {
                Calendar c = Calendar.getInstance();
                c.set(i, i1, i2);
                task.Deadline = c.getTimeInMillis(); //this is what you want to use later
            }
        });
        try {
            helper = new DataBaseHelper(getContext());
            db = helper.open();
        }
        catch(Exception e){
            Log.d(TAG, "onCreateView: " + e.getMessage());
        }
        return rootView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState){
        super.onViewCreated(view,savedInstanceState);
        Button imageButton=rootView.findViewById(R.id.PhotoButton);
        imageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent image=new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
                startActivityForResult(image,1);
            }
        });
        Button saveButton=rootView.findViewById(R.id.SaveButton);
        saveButton.setOnClickListener(new View.OnClickListener(){
           @Override
           public void onClick(View v){
               if(task.Photo==null){
                   Toast toast= Toast.makeText(getContext(),"Выберите картинку!",Toast.LENGTH_SHORT);
                   toast.show();
                   return;
               }
               if(task.Name.equals("") || task.Description.equals("")){
                   Toast toast=Toast.makeText(getContext(),"Заполните поля!",Toast.LENGTH_SHORT);
                   toast.show();
                   return;
               }
               Log.d("ExceptionLog", "Filler: " + String.valueOf(task.Deadline));

               ContentValues cv=new ContentValues();
               cv.put(helper.ColumnName,task.Name);
               cv.put(helper.ColumnDescription,task.Description);
               cv.put(helper.ColumnComplexity,task.ComplexityPosition);
               cv.put(helper.ColumnUrgency,task.UrgencyPosition);
               cv.put(helper.ColumnDeadline,task.Deadline);
               cv.put(helper.ColumnPhoto,task.Photo);
               CheckBox box=rootView.findViewById(R.id.Favorite);
               boolean a=false;
               if(box.isChecked()){
                    a=true;
               }
               cv.put(helper.ColumnFavorite,a);
               db.insert(helper.TABLE,null,cv);
           }
        });

    }
    @Override
    public void onActivityResult(int requestCode,int resultCode,Intent data){
        super.onActivityResult(requestCode,resultCode,data);
        if (data != null) {
            switch (requestCode) {
                case 1:
                    Uri selectedImage = data.getData();
                    String[] filePathColumn = {MediaStore.Images.Media.DATA};
                    Cursor cursor = getActivity().getContentResolver().query(selectedImage, filePathColumn, null, null, null);
                    cursor.moveToFirst();
                    int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
                    String imagepath = cursor.getString(columnIndex);
                    task.Photo=imagepath;
                    cursor.close();
                    break;
            }
        }
    }
}

File: ElevenLabSecondTry/app/src/main/java/com/example/elevenlabsecondtry/Fragments/addFragment.kt
package com.example.elevenlabsecondtry.Fragments

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.example.elevenlabsecondtry.ChessBoard
import com.example.elevenlabsecondtry.ChessPiece.*
import com.example.elevenlabsecondtry.R

class addFragment : Fragment() {

    private lateinit var rootView:View
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        rootView= inflater.inflate(R.layout.fragment_add, container, false)
        Filler()
        return rootView
    }
    private fun Filler(){
        var spinner: Spinner =rootView.findViewById(R.id.chessSpinner)
        var adapter=ArrayAdapter.createFromResource(rootView.context,R.array.Pieces,android.R.layout.simple_list_item_1)
        spinner.adapter=adapter;
        spinner=rootView.findViewById(R.id.ColorSpinner)
        adapter= ArrayAdapter.createFromResource(rootView.context,R.array.Colors,android.R.layout.simple_list_item_1)
        spinner.adapter=adapter
        val button:Button=rootView.findViewById(R.id.CreateButton)
        button.setOnClickListener(View.OnClickListener {
            Create()
        })
        val buttonauto:Button=rootView.findViewById(R.id.autoFill)
        buttonauto.setOnClickListener(View.OnClickListener {
            autoFill()
        })
    }
    private fun autoFill(){
        var piece:ChessPiece;
        piece=ChessPiece("K",'E',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("Q",'D',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("B",'C',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("B",'F',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("Kn",'G',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("Kn",'B',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("R",'A',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("R",'H',1,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'A',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'B',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'C',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'D',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'E',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'F',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'G',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'H',2,"White")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("K",'E',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("Q",'D',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("B",'C',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("B",'F',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("Kn",'G',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("Kn",'B',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("R",'A',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("R",'H',8,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'A',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'B',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'C',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'D',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'E',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'F',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'G',7,"Red")
        ChessBoard.Pieces.add(piece)
        piece=ChessPiece("P",'H',7,"Red")
        ChessBoard.Pieces.add(piece)
    }
    private fun Create(){
        var amountOfPieceFlag=true
        val piece:ChessPiece;
        val pieceSpinner: Spinner =rootView.findViewById(R.id.chessSpinner)
        val colorSpinner: Spinner =rootView.findViewById(R.id.ColorSpinner)
        val positionEditText=rootView.findViewById<EditText>(R.id.PositionEditText)
        val position:String=positionEditText.text.toString().uppercase()
        if(ChessBoard.Pieces.any { it.FullPosition==position}){
            Toast.makeText(rootView.context,"Позиция занята!",Toast.LENGTH_SHORT).show()
            return;
        }
        piece = when(pieceSpinner.selectedItem){
            "King"-> ChessPiece("K",position[0],position[1].digitToInt(),colorSpinner.selectedItem.toString())
            "Queen"-> ChessPiece("Q",position[0],position[1].digitToInt(),colorSpinner.selectedItem.toString())
            "Rook"-> ChessPiece("R",position[0],position[1].digitToInt(),colorSpinner.selectedItem.toString())
            "Bishop"-> ChessPiece("B",position[0],position[1].digitToInt(),colorSpinner.selectedItem.toString())
            "Knight"-> ChessPiece("Kn",position[0],position[1].digitToInt(),colorSpinner.selectedItem.toString())
            else-> ChessPiece("P",position[0],position[1].digitToInt(),colorSpinner.selectedItem.toString())
        }
        when(piece.Abbreviation){
            "K"->if(ChessBoard.Pieces.filter { it.Colour==piece.Colour && it.Abbreviation==piece.Abbreviation }.size>=1) amountOfPieceFlag=false
            "Q"->if(ChessBoard.Pieces.filter { it.Colour==piece.Colour && it.Abbreviation==piece.Abbreviation }.size>=1) amountOfPieceFlag=false
            "R"->if(ChessBoard.Pieces.filter { it.Colour==piece.Colour && it.Abbreviation==piece.Abbreviation }.size>=2) amountOfPieceFlag=false
            "B"->if(ChessBoard.Pieces.filter { it.Colour==piece.Colour && it.Abbreviation==piece.Abbreviation }.size>=2) amountOfPieceFlag=false
            "Kn"->if(ChessBoard.Pieces.filter { it.Colour==piece.Colour && it.Abbreviation==piece.Abbreviation }.size>=2) amountOfPieceFlag=false
            "P"->if(ChessBoard.Pieces.filter { it.Colour==piece.Colour && it.Abbreviation==piece.Abbreviation }.size>=8) amountOfPieceFlag=false
            else->amountOfPieceFlag=true
        }
        if(amountOfPieceFlag) {
            ChessBoard.Pieces.add(piece)
            piece.status.statusCheck()
        }
        else{
            Toast.makeText(rootView.context,"Нельзя добавить еще такую фигуру!",Toast.LENGTH_SHORT).show()
            return
        }
    }

}